doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,364 @@
+                # Step 1 and 2
+                os.environ["DATABRICKS_CONFIG_FILE"] = config["databricks_config_file"]
+                mlflow.set_tracking_uri("databricks")
+                mlflow.set_experiment_id(...)
+                mlflow.start_run()
 
-                wandb.init(
-                    id=context.get_trial_id(),
-                    name=context.get_trial_name(),
-                    group=context.get_experiment_name(),
-                    # ...
-                )
                 # ...
 
                 loss = optimize()
 
                 metrics = {"loss": loss}
-                # Only report the first worker results to wandb
-                if context.get_world_rank() == 0:
-                    wandb.log(metrics)
+                # Only report the results from the first worker to mlflow to avoid duplication
+
+                # Step 3
+                if train.get_context().get_world_rank() == 0:
+                    mlflow.log_metrics(metrics)
+
+.. tip::
+
+    A major difference between distributed and non-distributed training is that in distributed training, 
+    multiple processes are running in parallel and under certain setups they have the same results. If all 
+    of them are reported to the tracking backend, there may be duplicated results. To address that,  
+    Ray Train lets you apply logging logic to only the rank 0 worker with the following method:
+    :meth:`context.get_world_rank() <ray.train.context.TrainContext.get_world_rank>`.
+
+    .. code-block:: python
+
+        from ray import train
+        def train_func(config):
+            ...
+            if train.get_context().get_world_rank() == 0:
+                # do your logging logic only for rank0 worker.
+            ...
+
+The interaction with experiment tracking backend within the ``train_func`` can be broken 
+into 4 logical steps:
+
+- Set up to connect to a tracking backend
+- Configure and launch a run
+- Log
+- Finish the run
+
+Let's dive into each one of them.
+
+Step 1: Set up necessary components to connect to the tracking backend of your choice
+-------------------------------------------------------------------------------------
 
-                # Also report to Ray Train. Note that this _must_ happen for all workers.
-                train.report(metrics)
+First, you should choose which tracking backend to use: W&B, MLflow, TensorBoard etc.
 
-    .. tab-item:: PyTorch Lightning
+Some of them operate under either online or offline mode, each with different considerations when
+being set up.
+For online mode, you log towards a tracking service that is running. Usually you need credentials to access the service.
+Under this mode, you need to ensure that all nodes and worker processes have access to credentials.
+For offline mode, you log towards the local file directory. Usually no credentials are needed. You need to instead
+ensure that there is a shared file system where all nodes can write to.
```

Comment:
This can be condensed to just say you need to make sure you are authenticated properly on the distributed worker